mephisto/data_model/task.py

```diff
@@ -261,7 +261,12 @@ def get_valid_units_for_worker(self, worker: "Worker") -> List["Unit"]:
             is_self_set = map(lambda u: u.worker_id == worker.db_id, unit_set)
             if not any(is_self_set):
                 units += unit_set
-        valid_units = [u for u in units if u.get_assigned_agent() is None]
+        valid_units = [
+            u
+            for u in units
+            if u.get_assigned_agent() is None
```

Comment:
You don't need to check `u.get_assigned_agent()` being `None`, as `get_assigned_agent` will only ever return `None` for `Unit`'s that are in the `CREATED` and `LAUNCHED` statuses. As such, this check is redundant.